osquery/tables/system/darwin/system_extensions.mm

```diff
@@ -48,6 +48,62 @@ void getExtensionRow(const pt::ptree& extension, Row& r) {
   const auto category = extension.get_child("categories");
   r["category"] = getExtensionCategory(category);
   r["bundle_path"] = extension.get("container.bundlePath", "");
+  r["is_managed"] = INTEGER(0);
+}
+
+// com.apple.system-extension-policy payload defines the policy for
+// the system extensions. The function looks for team identifier or
+// extension bundle identifiers in the allowed list.
+// https://developer.apple.com/documentation/devicemanagement/systemextensions?language=objc
+void setPolicyFlag(const pt::ptree& ptree, Row& r) {
+  const auto& policies_attr_opt = ptree.get_child_optional("extensionPolicies");
+  if (!policies_attr_opt.has_value()) {
+    return;
+  }
+
+  const auto& policies_attr = policies_attr_opt.value();
+  const auto row_teamid = r["team"];
+  const auto row_identifier = r["identifier"];
+
+  for (const auto& policy : policies_attr) {
+    // Get the list of AllowedTeamIdentifiers that are valid and system
+    // extensions signed by them are allowed to load.
+    //
+    // All system extensions signed with any of the specified team identifiers
+    // will be considered to be approved
+    const auto& policy_item = policy.second;
+    const auto& allowed_team_opt =
+        policy_item.get_child_optional("allowedTeamIDs");
+    if (allowed_team_opt.has_value()) {
+      const auto& allowed_team = allowed_team_opt.value();
+      for (const auto& team : allowed_team) {
+        const auto teamid = team.second.get("", "");
+        if (teamid == row_teamid) {
+          r["is_managed"] = INTEGER(1);
+          break;
+        }
+      }
+    }
+
+    // get the list of system extensions that will always be approved
+    const auto& allowed_extensions_opt =
+        policy_item.get_child_optional("allowedExtensions");
+    if (allowed_extensions_opt.has_value()) {
```

Comment:
If system extensions can be explicitly allowed, can they also be explicitly denied?

If so, maybe this column should have three values (e.g., `allowed`, `unmanaged`, `denied`).
If not, maybe the column name or schema comments should clarify that this is just an `is_allowed` MDM setting.

Reply:
The MDM configuration can explicitly allow system extensions based on team identifiers, bundle identifiers, or both. It can also disable all system extensions other than those allowed in the config. But it can't deny extensions based on team identifier or bundle identifier. 

Also, system extensions denied will not show up in the list unless either the extension got installed before installing the MDM profile or if it is in the stale state from the older installation and waiting for the reboot to get removed. 

That's why I limited it to be in two-state: `managed` and `unmanaged`. I will update my comments to make them clear.

Reply:
I see. So there's a `disable all`, but that's not being captured by this table. This column won't be used to confirm hosts' allow/deny configuration, so much as it will confirm that the present system extensions are known/managed ones, and thus filter them out with a constraint.

Reply:
There is a `disable all` flag but the table does not capture it. I feel the `disable all` flag or `extension whitelisting` policies (irrespective of extensions installed) are more suited for another table showing the system extension payload/policies. Something on the line of [`manage_policies`](https://github.com/osquery/osquery/blob/master/specs/darwin/managed_policies.table) table.